lib/component.js

```diff
@@ -6,97 +6,89 @@ const inquirer = require('inquirer');
+const isDirectory = async source => {
+  const stats = await fs.promises.lstat(source);
+  return stats.isDirectory();
 };
 
-init();
-
-function init() {
-  const patternSrc = `${process.cwd()}/source/_patterns/`;
-  const patternDir = getDirectories(patternSrc);
+const getDirectories = async source => {
+  const directoryFiles = await fs.promises.readdir(source);
+  const directoryPaths = directoryFiles.map(name => path.join(source, name));
+  const isDirectoryResults = await Promise.all(directoryPaths.map(isDirectory));
+  return directoryPaths.filter((value, index) => isDirectoryResults[index]);
+};
 
+async function getMachineName() {
   const questions = [
     {
       type: 'input',
-      name: 'component_name',
+      name: 'componentName',
       message: 'What is the name your component?',
       filter: machineName,
     },
   ];
-  inquirer
-    .prompt(questions)
-    .then(answers => {
-      const componentName = answers.component_name.trim();
-      const defaultComponentTitle = humanName(componentName);
-      const questions = [
-        {
-          type: 'input',
-          name: 'component_title',
-          message: 'What is the human-readable title of your component?',
-          default: defaultComponentTitle,
-        },
-        {
-          type: 'list',
-          name: 'component_folder',
-          message: 'Component Location',
-          choices: patternDir.map(item => path.basename(item)),
-        },
-        {
-          type: 'input',
-          name: 'component_folder_sub',
-          message: 'Include subfolder or leave blank',
-        },
-      ];
-      return inquirer
-        .prompt(questions)
-        .then(answers =>
-          Object.assign(answers, { component_name: componentName })
-        );
-    })
-    .then(answers => {
-      const componentName = answers.component_name;
-      const componentTitle = answers.component_title;
-      const componentDocumentation = answers.document;
-      const componentLocation = path.join(
-        patternSrc,
-        answers.component_folder,
-        machineName(answers.component_folder_sub),
-        machineName(componentName)
-      );
-      const output = `---
+  const { componentName } = await inquirer.prompt(questions);
+  return componentName.trim();
+}
+
+async function getComponentDetails(componentName, patternDir) {
+  const defaultComponentTitle = humanName(componentName);
+  const detailedQuestions = [
+    {
+      type: 'input',
+      name: 'componentTitle',
+      message: 'What is the human-readable title of your component?',
+      default: defaultComponentTitle,
+    },
+    {
+      type: 'list',
+      name: 'componentFolder',
+      message: 'Component Location',
+      choices: patternDir.map(item => path.basename(item)),
+    },
+    {
+      type: 'input',
+      name: 'componentFolderSub',
+      message: 'Include subfolder or leave blank',
+    },
+  ];
+  return await inquirer.prompt(detailedQuestions);
```

Comment:
If you're just returning a promise, you don't need to `await` it here (under the hood, it forces two yields: the first is the `await` inside this function, followed by any callers' `await`).